Frontend/android-native/app/src/main/java/in/juspay/mobility/CommonJsInterface.java

```diff
@@ -1930,6 +1931,58 @@ public void run() {
         });
     }
 
+    @JavascriptInterface
+    public void setScaleType (String id, String imageUrl, String scaleType){
+        if (activity == null) return;
+        activity.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                if (id != null || imageUrl != null){
+                    ImageView imageView = activity.findViewById(Integer.parseInt(id));
+                    new Thread(new Runnable() {
+                        @Override
+                        public void run() {
+                            try {
+                                URL url = new URL(imageUrl);
+                                Bitmap bitmap = BitmapFactory.decodeStream(url.openConnection().getInputStream());
+                                Handler mainLooper = new Handler(Looper.getMainLooper());
+                                mainLooper.post(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        if (bitmap == null) return;
+                                        int tempWidth = Math.min(bitmap.getWidth(), getScreenWidth());
+                                        imageView.getLayoutParams().height =  (tempWidth * (getScreenHeight()/getScreenWidth()));
+                                        imageView.setScaleType(getScaleTypes(scaleType));
+                                        imageView.setImageBitmap(bitmap);
+                                        LinearLayout linearLayout = (LinearLayout) imageView.getParent();
+                                        linearLayout.removeAllViews();
+                                        linearLayout.addView(imageView);
+                                        imageView.setVisibility(View.VISIBLE);
+                                    }
+                                });
+                            } catch (Exception e) {
+                                throw new RuntimeException(e);
+                            }
+                        }
+                    }).start();
+                }
+            }
+        });
+    }
```

Comment:
These many threads are not required, remove unwanted threads. Your code can be reduced 

`  @JavascriptInterface
    public void setScaleType (String id, String imageUrl, String scaleType){
        if (activity == null) return;
        if (id != null || imageUrl != null){
            ImageView imageView = activity.findViewById(Integer.parseInt(id));
            try {
                URL url = new URL(imageUrl);
                Bitmap bitmap = BitmapFactory.decodeStream(url.openConnection().getInputStream());
                Handler mainLooper = new Handler(Looper.getMainLooper());
                mainLooper.post(() -> {
                    if (bitmap == null) return;
                    int tempWidth = Math.min(bitmap.getWidth(), getScreenWidth());
                    imageView.getLayoutParams().height =  (tempWidth * (getScreenHeight()/getScreenWidth()));
                    imageView.setScaleType(getScaleTypes(scaleType));
                    imageView.setImageBitmap(bitmap);
                    LinearLayout linearLayout = (LinearLayout) imageView.getParent();
                    linearLayout.removeAllViews();
                    linearLayout.addView(imageView);
                    imageView.setVisibility(View.VISIBLE);
                });
            } catch (Exception e) {
                throw new RuntimeException(e);
            }
        }
    }`